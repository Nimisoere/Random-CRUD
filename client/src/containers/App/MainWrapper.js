import React, { memo } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { ThemeProps } from "../../shared/prop-types/ReducerProps"

export const MainWrapper = memo(props => {
  const { theme, children } = props;

  return (
    <div className={theme.className}>
      <div className="wrapper">{children}</div>
    </div>
  );
});

MainWrapper.propTypes = {
  theme: ThemeProps.isRequired,
  children: PropTypes.element.isRequired
};

export default connect(state => ({
  theme: state.theme
}))(MainWrapper);
