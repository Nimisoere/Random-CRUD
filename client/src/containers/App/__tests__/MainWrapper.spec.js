import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { MainWrapper } from "../MainWrapper";

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    theme: { className: "theme-light" },
    children: React.createElement("div")
  };
  const enzymeWrapper = shallow(<MainWrapper {...props} />);
  return {
    props,
    enzymeWrapper
  };
}

describe("components", () => {
  describe("MainWrapper", () => {
    it("Match Snapshot", () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
