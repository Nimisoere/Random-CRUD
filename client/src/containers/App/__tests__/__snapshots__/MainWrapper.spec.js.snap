// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components MainWrapper Match Snapshot 1`] = `ShallowWrapper {}`;
